spec/lib/whitehall_importer/create_revision_spec.rb

```diff
@@ -181,6 +181,20 @@
         .to eq(lead_organisation["content_id"])
     end
 
+    it "sets the first lead organisation as the primary publishing organisation" do
+      first_lead_organisation = build(:whitehall_export_organisation, :lead)
+      second_lead_organisation = build(:whitehall_export_organisation, :lead, lead_ordering: 2)
+      whitehall_edition = build(
+        :whitehall_export_edition,
+        organisations: [first_lead_organisation, second_lead_organisation],
+      )
+
+      revision = described_class.call(document_import, whitehall_edition)
+
+      expect(revision.primary_publishing_organisation_id)
+        .to eq(first_lead_organisation["content_id"])
+    end
+
     it "sets supporting organisations" do
       lead_organisation = build(:whitehall_export_organisation, :lead)
       supporting_organisation = build(:whitehall_export_organisation)
```

Comment:
I don't think we're testing the scenario of there being multiple lead organisations.  The second lead organisation has lead=true, so I believe https://github.com/alphagov/content-publisher/pull/1665/commits/a6e5dd9b6274c6cac65c82023be7f253e6b72bbd#diff-eb91f6f587e00528b32cc8dc205dcd0bL109 would mean this doesn't get added as a supporting organisation either.  Maybe this test could be updated (or another added) to reflect this scenario?

Reply:
You're absolutely right Bruce, I've created a context for multiple lead orgs and two unit tests off of your suggestion to better describe what each test is actually testing instead of pushing them into one.